dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -320,6 +320,12 @@ void MPPTask::preprocess()
     auto start_time = Clock::now();
     initExchangeReceivers();
     executeQuery(*context);
+
+    // Do set-up work for tunnels and receivers after ProcessListEntry is constructed,
+    // so that we can propagate current_memory_tracker into them.
+    tunnel_set->updateMemTracker();
+    receiver_set->setUpConnection();
```

Comment:
`executeQuery` will do learner read, which may take a long time, set up connection after learner read may cause write timeout, this is what we try to avoid. Ref to comments here: https://github.com/pingcap/tiflash/pull/1360/files#diff-cdedd6c3fc4f78c24656df759570ba8232f4167f19fa457df6b2741a6ed00293R82

Reply:
so we need revert the implementation right after the ProcListEntry is created?

Reply:
fixed